### DISCUSSION TITLE
at property not working

### DISCUSSION THREAD

**Discussion by liviacorradini** [GrapesJS/grapesjs#5220] [Q&A]
Hello, I'm trying to add a component before or after the selected component. But even using at the new component always goes to the end of all.

Code example to add after: 

let selected = editor.getSelected();
  let index = selected.index();  
  editor.getomponents().append(
  '<div> example </div>',{at: index+1})

Could you tell me what could be wrong with the code? Thanks

**Comment by artf** [comment]
Can you provide a reproducible demo of your use case? 
It doesn't look like you're using the proper API, `getomponents`? I guess it's `getComponents` but even in that case there is no `append` method...

**Comment by liviacorradini** [comment]
Demo of use: 
![image](https://github.com/GrapesJS/grapesjs/assets/97858789/4c9146e2-af70-43a2-9365-ae8f37d5487d)
![screen-capture](https://github.com/GrapesJS/grapesjs/assets/97858789/36dc3105-bfb8-43fa-b455-776026086482)

**Reply by artf** [reply]
don't use `getComponents().add`, it's not a public API. 
I'd recommend this for your use case:
```js
const selected = editor.getSelected();
const parent = selected.parent();
parent.append('...', { at: selected.index() + 1  })
```
